sklearn/discriminant_analysis.py

```diff
@@ -71,14 +79,8 @@ def _cov(X, shrinkage=None, covariance_estimator=None):
                 s = sc.scale_[:, np.newaxis] * s * sc.scale_[np.newaxis, :]
             elif shrinkage == "empirical":
                 s = empirical_covariance(X)
-            else:
```

Comment:
Since here we have more options than in the estimator class, I think that we can keep the current checks. It will be replace when we will validate the function itself.

Reply:
this is a private function, so we probably won't validate its params. Since this function is only called by the estimator I think that this check can be removed. It should be unreachable code now